### PR TITLE
GRIDEDIT-540_error_handling

### PR DESCRIPTION
**Pull request by ahmad-el-sayed**

### CODE REVIEW

File: meshkernel/meshkernel.py [hunk 1]
```diff
@@ -88,16 +81,69 @@ def __init__(self, is_geographic: bool = False):
         elif system == "Darwin":
             lib_path = os.path.join(file_path, "libMeshKernelApi.dylib")
         else:
-            if not str:
+            if not system:
                 system = "Unknown OS"
             raise OSError("Unsupported operating system: {}".format(system))
 
         self.lib = CDLL(str(lib_path))
+        self.ExitCode = self.__get_exit_codes()
```
Comment (lucacarniato) [inline meshkernel/meshkernel.py:89]: Sonar Cloud complains about ExitCode name. Should it be   self.exit_code ?

File: meshkernel/py_structures.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 import numpy as np
 from numpy import ndarray
 
-from meshkernel.errors import InputError
+import meshkernel.errors as mkerr
```
Comment (lucacarniato) [inline meshkernel/py_structures.py:9]: as mk_errors?

File: setup.py [hunk 3]
```diff
@@ -77,7 +77,7 @@ def get_library_meta() -> LibraryMeta:
     elif system == "Darwin":
         name = "libMeshKernelApi.dylib"
     else:
-        if not str:
+        if not system:
```
Comment (lucacarniato) [inline setup.py:80]: :-)

**Review by lucacarniato — APPROVED** [review]

File: setup.py (hunk 3 above)
Comment (ahmad-el-sayed) [reply setup.py:80]: Already fixed in master. See: [setup.py](https://github.com/Deltares/MeshKernelPy/blob/66de45b6ad8f31fbe17a83d133b1e537aa54042d/setup.py#L81C23-L81C23) and [meshkernel/meshkernel.py](https://github.com/Deltares/MeshKernelPy/blob/66de45b6ad8f31fbe17a83d133b1e537aa54042d/meshkernel/meshkernel.py#L92).

File: meshkernel/py_structures.py (hunk 2 above)
Comment (ahmad-el-sayed) [reply meshkernel/py_structures.py:9]: Sure.

File: meshkernel/meshkernel.py (hunk 1 above)
Comment (ahmad-el-sayed) [reply meshkernel/meshkernel.py:89]: done

**Pull request merged**